File: Funcionarios Java/FuncionariosExercicio/src/Teste.java
import java.util.Scanner;

public class Teste {
    public static void main(int num_funcionarios){
        Scanner scan = new Scanner(System.in);
        String CPFentered;
        int option;
        double salario;
        float comission;
        int somaTotal = 0;
        String Nome;
        Funcionario[] arrayFunc = new Funcionario[num_funcionarios];
        for(int i = 0; i < num_funcionarios; i++ ){
            System.out.println("Insira o numero relativo ao cargo:\n 1) Gerente\n 2) Assistente\n 3) Vendedor");
            option = scan.nextInt();
            System.out.println("Insira o CPF no formato (XXXXXXXXXXX):");
            scan.nextLine();
            CPFentered = scan.nextLine();


            if(!Funcionario.verificaCPF(CPFentered)){
                System.out.println("CPF invalido, o cadastro será reiniciado");
                continue;
            }

            System.out.println("Insira o nome:");
            Nome = scan.next();
            System.out.println("Insira o salario base:");
            salario = scan.nextDouble();
            switch(option){
                case 1:
                    arrayFunc[i] = new Gerente();
                    somaTotal += arrayFunc[i].calculaSalario(salario);
                    arrayFunc[i].CPF = CPFentered;
                    arrayFunc[i].nome = Nome;
                    break;
                case 2:
                    arrayFunc[i] = new Assistente();
                    somaTotal += arrayFunc[i].calculaSalario(salario);
                    arrayFunc[i].CPF = CPFentered;
                    arrayFunc[i].nome = Nome;
                    break;
                case 3:
                    System.out.println("Insira a comissao:");
                    comission = scan.nextFloat();
                    arrayFunc[i] = new Vendedor(comission);
                    somaTotal += arrayFunc[i].calculaSalario(salario);
                    arrayFunc[i].CPF = CPFentered;
                    arrayFunc[i].nome = Nome;
                    break;
            }
            System.out.println("Despesas totais da folha salarial da empresa: " + somaTotal);



        }


    }
}

File: Funcionarios Python/Teste.py
from Funcionario import Funcionario
from Gerente import Gerente
from Assistente import Assistente
from Vendedor import Vendedor


class Teste:
    def testeMain(self, num_funcionarios):

        option = 0
        salario = 0.0
        comission = 0.0
        somaTotal = 0
        nome = ''
        CPFentered = ''
        arrayFunc = [num_funcionarios]

        for i in range(num_funcionarios):
            option = int(input("Insira o numero relativo ao cargo:\n 1) Gerente\n 2) Assistente\n 3) Vendedor\n"))
            CPFentered = input("Insira o CPF no formato (XXXXXXXXXXX): ")
            if not Gerente().verificaCPF(CPFentered):
                print("CPF invalido, o cadastro será reiniciado")
                continue
            nome = input("Insira o nome: ")
            salario = float(input("Insira o salario base: "))
            match option:
                case 1:
                    arrayFunc[i] = Gerente()
                    somaTotal += arrayFunc[i].calculaSalario(salario)
                    arrayFunc[i].CPF = CPFentered
                    arrayFunc[i].nome = nome
                case 2:
                    arrayFunc[i] = Assistente()
                    somaTotal += arrayFunc[i].calculaSalario(salario)
                    arrayFunc[i].CPF = CPFentered
                    arrayFunc[i].nome = nome
                case 3:
                    comission = float(input("Insira a comissão: "))
                    arrayFunc[i] = Vendedor(comission)
                    somaTotal += arrayFunc[i].calculaSalario(salario)
                    arrayFunc[i].CPF = CPFentered
                    arrayFunc[i].nome = nome

        print("Despesas totais da folha salarial da empresa: " + str(somaTotal))



